### PR TITLE
payload tracker requires request_id with no dashes

### PR DESCRIPTION
**Pull request by thearifismail**
# Overview

This PR is being created to address the issue reported in [ESSNTL-4044](https://issues.redhat.com/browse/ESSNTL-4044).
It also removes dashes from request_id before sending a message to Payload-tracker.  Payload tracker can not handle dashes in request_ids and the message gets lost without providing any information.  

A more permanent fix should be provided by `Payload Tracker` to handle `request_ids` with or without dashes as both formats are valid.  Host inventory uses dashes in request ID.

## PR Checklist

- [ ] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [ ] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [ ] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

**Comment by fstavela** [comment]
@thearifismail pre-commit failed

**Comment by thearifismail** [comment]
@fstavela Now an `iqe-test` failing.  The failing test is `test_delete_bulk_hostname_or_id`. Can you please take a look at it?

**Review by fstavela — APPROVED** [review]

File: app/payload_tracker/__init__.py [hunk 1]
```diff
@@ -185,6 +185,7 @@ def _send_message(self, message):
 
         try:
             self._producer.produce(self._topic, message.encode("utf-8"))
+            self._producer.poll(0)
```
Comment (kruai) [inline app/payload_tracker/__init__.py:188]: Is this required bc of confluent-kafka?

**Review by kruai — APPROVED** [review]
Just a Q but LGTM

**Pull request merged**